.github/workflows/python-unit-tests.yml

```diff
@@ -0,0 +1,45 @@
+name: Python Unit Tests
```

Comment:
probably also better to add a new job in `unit-tests.yml` than to create a new workflow

Reply:
Aside from being tests, I wouldn't say these tests are related to the unit tests, nor do they share any build artifacts. I personally feel that it would be more helpful and clear to separate them, but I am also fine with running them with the other unit tests if you still think so.